eng/tox/create_whl_for_dependencies.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# This script creates wheels for dependencies of a package and installs them.
+
+import subprocess
+import argparse
+import os
+import logging
+import sys
+import fnmatch
+import tempfile
+from packaging.version import parse
+from ci_tools.parsing import ParsedSetup
+from ci_tools.build import create_package
+from ci_tools.variables import in_ci
+logging.getLogger().setLevel(logging.INFO)
+
+
+def install_dev_reqs(file, pkg_root, temp_dir):
```

Comment:
How close is this code to original? Worth the trouble to move under `ci_tools.functions` or a like namespace?

That way we can duplicate this code exactly once, then share in both locations.

Reply:
Yeah and when we move the common code, please type it + code comment description. No reason to do `param` strings if you don't wish, but the description + type inline help a ton when using this function later.

Reply:
The main differences with this code are 1) we don't overwrite the user's dev_requirements.txt when running locally and 2) we build _and install_ the whls. 

So we could update the original functions to only do those two things when not running in CI, move them to ci_tools.functions, and then share them across this script and the tox harness. Would you like me to do that?